New Dataset for R2 added

siddharth0248 opened the pull request:
## Add new dataset for R2
-Creating mdx for new datasets (MiCASA and LPJ)
-Testing the map explore
-Adding new content

siddharth0248 commented:
@slesaad @j08lue can one of you point out why titiler increasing the spatial resolution of the LPJ data. I tried to visualize the data in QGIS and in our portal. Any variable in mdx that is doing this

![image](https://github.com/US-GHG-Center/veda-config-ghg/assets/118487103/9c5d4fe8-9df4-4d99-804c-48dbf20d46f5)

![image](https://github.com/US-GHG-Center/veda-config-ghg/assets/118487103/820f2a15-5970-4ab4-899f-b09b8573fcb6)

slesaad commented:
@siddharth0248 we had this same issue a while ago, which was fixed by pinning rio_tiler to a post release that @vincentsarago created https://github.com/US-GHG-Center/ghgc-backend/pull/59/files

we recently removed that pin because  I understood that the fix was incorporated in the new releases (see: https://github.com/NASA-IMPACT/veda-backend/pull/267/files).

@vincentsarago these are the requirements used for this deployment - https://github.com/US-GHG-Center/ghgc-backend/blob/dev/raster_api/runtime/setup.py. any idea what's up?

j08lue commented:
Preview to debug the issue: https://deploy-preview-333--ghg-demo.netlify.app/data-catalog/lpjwsl-wetlandch4-grid-v2/explore?projection=equirectangular%7C%7C&basemapid=satellite&datetime=1990-02-05T00%3A00%3A00.000Z

vincentsarago commented:
> @vincentsarago these are the requirements used for this deployment - https://github.com/US-GHG-Center/ghgc-backend/blob/dev/raster_api/runtime/setup.py. any idea what's up?

well, as far I can see, this shows titiler.core requirement `"titiler.core>=0.15.5,<0.16"` which should depends on rio-tiler >6.2.5 (https://github.com/cogeotiff/rio-tiler/issues/654 was fixed in 6.2.7), I don't see why pip won't install rio-tiler 6.4.5 (latest version)

siddharth0248 commented:
@vincentsarago @j08lue when can we get this resolved ? or any pointers for me ?

j08lue commented:
> when can we get this resolved ? or any pointers for me ?

As far as I understand, the issue with the rendering is not related to this PR or the files, but only the rio-tiler version in the dev stage of our data services, which for some reason is out of date.

Therefore, you should in principle be ok to merge this PR and the data shoud look right in production.

I would like to verify this first, though, but it looks like the production tiler does not have access to the dev bucket? This does not load:

https://ghg.center/api/raster/cog/map?url=s3://ghgc-data-store-dev/lpjwsl-wetlandch4-daygrid-v2/ensemble_mean/LPJ_EOSIM_L2_CH4e_dch4e_ensemble_mean_19900101day.tif&colormap_name=magma&rescale=0%2C2&nodata=0

and this throws access denied:

https://ghg.center/api/raster/cog/info?url=s3://ghgc-data-store-dev/lpjwsl-wetlandch4-daygrid-v2/ensemble_mean/LPJ_EOSIM_L2_CH4e_dch4e_ensemble_mean_19900101day.tif&colormap_name=magma&rescale=0%2C2&nodata=0

So we would need to fix the access (@slesaad or @amarouane-ABDELHAK). Or were you planning on moving the data to a production bucket, @siddharth0248, and could fast-forward that?

siddharth0248 commented:
Changed the min max values for each of the layer. This helps in distinguishing high and super high values. eg, USA and China

Original/Proposed values in document
![image](https://github.com/US-GHG-Center/veda-config-ghg/assets/118487103/5a3147c4-1560-4013-96fd-21f766b2e7a6)

Changed values
<img width="1255" alt="image" src="https://github.com/US-GHG-Center/veda-config-ghg/assets/118487103/3da2e4c3-23b0-4a60-83e1-013c558dee7d">

@deborahUAH please have a look

vincentsarago commented:
After talking with @siddharth0248 and @SwordSaintLancelot we found that the dataset had a weird transform (origin of the file being lower right, instead of top right). For some Reason QGIS is able to work with it, but rio-tiler couldn't. 

I've added a fix in rio-tiler https://github.com/cogeotiff/rio-tiler/pull/696 which I hope to get merged/released today

deborahUAH commented:
@siddharth0248 regarding images above about changing visualization scale.  NBE is a new layer right?  Confirm choice of color ramp and min/max for visualization with data provider.

siddharth0248 commented:
The content has been checked by the team.

j08lue reviewed (changes requested):
This changes all datasets, not just LPJ and MiCASA as it says in the title.

I reviewed the data (exploration) only for LPJ and MiCASA.

LPJ looks right.

MiCASA is only available for 2001. Is that expected?

Dataset overview pages look right - nice improvement with the new layout.

siddharth0248 commented on .env:
```diff
@@ -8,10 +8,10 @@ APP_DESCRIPTION='Exploring Greenhouse Gas Data; Driving Sustainable Strategies t
 
 # Endpoint for the STAC server. No trailing slash.
-API_STAC_ENDPOINT='https://ghg.center/api/stac'
+API_STAC_ENDPOINT='https://dev.ghg.center/ghgcenter/api/stac'
```
can you check the changes I made now @j08lue

j08lue replied:
```suggestion
API_STAC_ENDPOINT='https://ghg.center/api/stac'
```

siddharth0248 commented on .env:
```diff
@@ -8,10 +8,10 @@ APP_DESCRIPTION='Exploring Greenhouse Gas Data; Driving Sustainable Strategies t
 
 # Endpoint for the Tiler server. No trailing slash.
-API_RASTER_ENDPOINT='https://ghg.center/api/raster'
+API_RASTER_ENDPOINT='https://dev.ghg.center/ghgcenter/api/raster'
```
Since MiCASA is daily file we haven't pushed all the data in dev

j08lue replied:
I see. But for merging this PR to `develop` (which is our staging for prod), should we change the endpoint back to prod?

siddharth0248 replied:
Explore map won't load any layer if API_STAC_ENDPOINT='https://ghg.center/ghgcenter/api/stac'. Data is still in dev and not in prod

j08lue replied:
```suggestion
API_RASTER_ENDPOINT='https://ghg.center/api/raster'
```

j08lue reviewed (changes requested):
Btw, do you mean to change the `.veda/ui` submodule checkpoint? Not sure which version you are changing it to here and if it is required for this PR?

hanbyul-here commented:
I have the same comments with @j08lue , If the data is not on prod endpoint, it means that it is not ready to be deployed to production. Please change the endpoint to prod when it is ready.

j08lue commented:
@siddharth0248 please check the ECCO Overview page - there was a merge conflict and I am not sure I kept the right edit. Looks like this now:

![image](https://github.com/US-GHG-Center/veda-config-ghg/assets/3404817/b35dd909-a8ce-4600-8978-57fe4f5a1273)

j08lue commented:
MiCasa also still needs the additional layer info for E&A:

![image](https://github.com/US-GHG-Center/veda-config-ghg/assets/3404817/8822eb57-d10b-448b-aacd-7974a9700fd3)

j08lue commented:
> I haven't added the media section, because how don't know how you grabbed it

Yeah, @faustoperez made the thumbnails last time

- https://github.com/US-GHG-Center/veda-config-ghg/pull/317

I think you can just take a screen grab from the rendered map of whichever extent and zoom level best brings out the characteristics of the dataset. You could also use the `/raster/cog/preview` endpoint of the tiler, if you anyways want a global coverage thumbnail, e.g.: https://dev.ghg.center/ghgcenter/api/raster/cog/preview?url=s3://ghgc-data-store-dev/MiCASA/v1/daily/2001/01/MiCASAv1_flux_NBE_x3600_y1800_daily_20010101.tif&colormap_name=purd&rescale=0%2C1.5

j08lue reviewed (approved):
Looks good and complete now, apart from the missing layer thumbnails for MiCASA and LPj. 

They have the fall-back to the dataset thumbnail, though, so that is not a blocker for merging.

![image](https://github.com/US-GHG-Center/veda-config-ghg/assets/3404817/8cc4481f-0c3f-42ee-b3ed-c7c5594fed10)
